tests/models/test_horovod.py

```diff
@@ -78,7 +78,6 @@ def test_horovod_cpu(tmpdir):
         limit_train_batches=0.4,
         limit_val_batches=0.2,
         accelerator="horovod",
-        deterministic=True,
```

Comment:
note to reviewer: these started failing because the BoringModel is not deterministic

```
[1]<stderr>:RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
```

Reply:
Are you sure the model is the problem? The error message says:

> To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility

Couldn't you just set the environment variable?

Reply:
we could set this directly in our CI configuration maybe. not sure.

they also write "This behavior is expected to change in a future release" in that section of the docs

Reply:
> they also write "This behavior is expected to change in a future release" in that section of the docs

This is the main reason I didn't enable this because I'm not sure how future-proof this is, and if one day they remove this then all of these tests will suddenly start failing. These tests aren't reliant on determinism so I wanted to remove one variable/possible source of failure from them.